### PR TITLE
Trust wallet integration & SDK

### PR DESCRIPTION
**Pull request by karim-yassine**
Hey web3-onboard team! I'm Karim, and I'm representing the Web Extension team of [Trust Wallet](https://trustwallet.com/). Based on [Pull Request Guide](https://github.com/blocknative/web3-onboard/blob/v2-web3-onboard-develop/pull_request_template.md), I'm opening this PR with a complete list of the changes it introduces (all related to Trust Wallet).

### Description
- Trust Wallet SDK package was created.
- Not displaying Trust Wallet name and logo when wallet was injected fixed.
- Conflicts between multiple injected wallets fixed.
- Documentation with Trust Wallet examples added.

### Checklist
- [x] The version field in `package.json` of the package you have made changes in is incremented following [semantic versioning](https://semver.org/) and using alpha release tagging
- [x] The box that allows repo maintainers to update this PR is checked
- [x] I tested locally to make sure this feature/fix works
- [x] I have run `yarn file-check`, `yarn type-check` & `yarn build` to confirm there are not any associated errors
- [x] This PR passes the Circle CI checks

#### Tests with demo app (injected)
- [x] send transaction
- [x] switch chains
- [x] sign message
- [x] sign typed message
- [x] disconnect

#### Tests with demo app (SDK)
- [x] send transaction
- [x] switch chains
- [x] sign message
- [x] sign typed message
- [x] disconnect

#### 
- [x] QA (ongoing)

### CODE REVIEW

File: packages/trust/package.json
Comment (Adamj1232) [inline packages/trust/package.json:3]:
```suggestion
  "version": "2.0.0-alpha.1",
```
For staging testing purposes prior to public package release

File: packages/trust/README.md
Comment (Adamj1232) [inline packages/trust/README.md:4]:
```suggestion

Trust Wallet SDK wallet module for connecting to Web3-Onboard. Web3-Onboard makes it simple to connect Ethereum hardware and software wallets to your dapp. Features standardized spec compliant web3 providers for all supported wallets, framework agnostic modern javascript UI with code splitting, CSS customization, multi-chain and multi-account support, reactive wallet state subscriptions and real-time transaction state change notifications.

```

File: packages/trust/package.json
Comment (Adamj1232) [inline packages/trust/package.json:4]:
```suggestion
  "description": "Trust Wallet SDK wallet module for connecting to Web3-Onboard. Web3-Onboard makes it simple to connect Ethereum hardware and software wallets to your dapp. Features standardized spec compliant web3 providers for all supported wallets, framework agnostic modern javascript UI with code splitting, CSS customization, multi-chain and multi-account support, reactive wallet state subscriptions and real-time transaction state change notifications.",
```

**Comment by Adamj1232** [comment]
Thanks for putting this PR together @karim-yassine ! I left a few comments to address. Can you test and complete the new wallet injected and SDK checklist when you have a moment? Thanks! Looking forward to adding support!

**Comment by karim-yassine** [comment]
Hello @Adamj1232, all the changes suggested where committed!
Our QA team is working on testing the wallet integration; could you please approve the Vercel deployment workflow?
Thank you in advance!

**Comment by Adamj1232** [comment]
@karim-yassine the vercel deployment is for our docs site we pulled into the develop branch. Please pull the latest develop when you have a chance.
For a vercel demo app of the Demo app for QA-ing wallet integration you will have to run `yarn build` within the demo package and then upload the public file from the demo package to vercel manually. We ill have this automated in the future.

**Comment by karim-yassine** [comment]
@Adamj1232 awesome, I'll update you when QA and the checklist are completed.
Thanks again Adam!

**Comment by Adamj1232** [comment]
@karim-yassine friendly ping to see how things are going and a reminder to pull in the latest develop and handle conflicts.
If there is anything I can do to help please feel free to reach out and let me know 😄

**Comment by karim-yassine** [comment]
@Adamj1232 Thanks for reaching out!

We are still under QA; we are fixing some small edge cases we managed to find. I'll make sure to sync the brach with develop and fix the conflicts before asking for new review and merge.

Everything should be done and ready for merge by the end of this week; I will keep you informed :)

**Comment by Adamj1232** [comment]
@karim-yassine we are going to be putting together a release candidate today. Any chance this PR is ready for review? If not we will have a following release coming out next week (or the week after)

**Comment by karim-yassine** [comment]
@Adamj1232 Hello! I'll be opening PR today, we found a bug in the index of injected package that we fixed, I'm just finalizing testing and making sure that it works correctly with other wallets.

**Comment by karim-yassine** [comment]
@Adamj1232 ready for review.

Regarding our fix in the index file of injected package, the bug here was that the implementation assumed that if there is a provider array, that should only be checked, which is wrong because it excludes the global scope. For that reason we first check the global scope and then the providers array, if exist.

**Comment by Adamj1232** [comment]
@karim-yassine thank you for the PR! I will test now! Can you give me write access if not already so I can add docs for your new module integration?

**Review by Adamj1232 — APPROVED** [review]
Thank you again for the PR and collaboration! Really appreciate it!

**Comment by Adamj1232** [comment]
@karim-yassine If interested in doing a bit of co-marketing/tweeting on the new support and package please feel free to reach out 😃

**Comment by atifnimran** [comment]
Please ok

**Pull request merged**
